src/OrleansEventSourcing/EventStorage/LogViewAdaptor.cs

```diff
@@ -0,0 +1,342 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Orleans.EventSourcing.Common;
+using Orleans.LogConsistency;
+using Orleans.Runtime;
+
+
+namespace Orleans.EventSourcing.EventStorage
+{
+
+    /// <summary>
+    /// A log view adaptor that is based on an event store
+    /// </summary>
+    /// <typeparam name="TState">The type of the log view, or grain state.</typeparam>
+    /// <typeparam name="TEvent">The type of the events (usually just object)</typeparam>
+    class EventStoreLogViewAdaptor<TState, TEvent> : PrimaryBasedLogViewAdaptor<TState, TEvent, SubmissionEntryWithGuid<TEvent>>
+        where TState : class, new() where TEvent : class
+    {
+
+        public EventStoreLogViewAdaptor(
+            ILogViewAdaptorHost<TState, TEvent> host, 
+            ILogConsistencyProvider provider,
+            TState initialstate, 
+            ILogConsistencyProtocolServices services)  : base(host, initialstate, services)
+        {
+            this.provider = provider as EventStorageProviderBase;
+
+            if (this.provider == null)
+                throw new ArgumentException("provider must be an event store", nameof(provider));
+
+        }
+
+
+        private EventStorageProviderBase provider;
+
+        private TState ConfirmedStateInternal;
+        private int ConfirmedVersionInternal;
+
+        // streamname must be constructed lazily because grain reference is not available at construction time
+        private string streamName
+        {
+            get
+            {
+                return _streamName ??
+                    (_streamName = provider.DefaultStreamName(Host.GetType(), Services.GrainReference));
+            }
+        }
+        private string _streamName;
+
+        protected override TState LastConfirmedView()
+        {
+            return ConfirmedStateInternal;
+        }
+
+        protected override int GetConfirmedVersion()
+        {
+            return ConfirmedVersionInternal;
+        }
+
+        protected override void InitializeConfirmedView(TState initialstate)
+        {
+            ConfirmedStateInternal = initialstate;
+            ConfirmedVersionInternal = 0;
+        }
+
+        public override async Task<IReadOnlyList<TEvent>> RetrieveLogSegment(int fromVersion, int toVersion)
```

Comment:
I think it is better to keep it mandatory here, this way there is no confusion about how synchronization and consistency of the observed version (e.g. how the returned log version relates to the current grain state version and to the timing of the notification calls of OnStateChanged). 

It easy and clear to just explicitly pass `Version` (a getter for the latest version number) as the toParameter.